src/Lecture3.hs

```diff
@@ -125,9 +133,11 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
+  chest1 <> chest2 = Reward ((rewardGold chest1) <> (rewardGold chest2)) ((rewardSpecial chest1) || (rewardSpecial chest2))
 
 
 instance Monoid Reward where
+  mempty = Reward (Gold 0) False
```

Comment:
You can use `mempty` for `Gold` here because it already has a `Monoid` instance 🏅 
```suggestion
  mempty = Reward mempty False
```